src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -772,26 +750,46 @@ public static String getNodeHostname(@Nullable EnvVars envVars, @Nullable Comput
                 Node node = computer.getNode();
                 if (node != null) {
                     FilePath rootPath = node.getRootPath();
-                    if (isUnix(rootPath)) {
-                        ShellCommandCallable hostnameCommand = new ShellCommandCallable(
-                                Collections.emptyMap(), HOSTNAME_CMD_TIMEOUT_MILLIS, "hostname", "-f");
+                    if (rootPath != null) {
+                        String[] command;
+                        if (isUnix(rootPath)) {
+                            command = new String[]{"hostname", "-f"};
+                        } else {
+                            command = new String[]{"hostname"};
```

Comment:
Does this command work in all Windows environments? If not, what would happen in that case? Would it throw an exception?

Reply:
Not sure about every possible version, but it does seem to be a standard command. If it doesn't work an exception will be thrown, then caught in this method and logged. `null` hostname will be returned. Given that this is the last possible fallback, seems like an acceptable outcome